opam lint fails with version 0.2.0

I get the following error with the latest release and `opam 1.2.2`:

```
Cloning into '_build/gsl-1.22.0.build'...
done.
Wrote archive _build/gsl-1.22.0.tbz

Linting distrib in _build/gsl-1.22.0
[FAIL] lint opam file gsl.opam:
       opam lint --warn=-21-32-48 -s messages:
       opam: unknown option `--warn'.
       Usage: opam lint [OPTION]... [FILE]
       Try `opam lint --help' or `opam --help' for more information.
[FAIL] lint _build/gsl-1.22.0 failure: 1 errors.

Building package in _build/gsl-1.22.0
[ OK ] package builds

Running package tests in _build/gsl-1.22.0
[ OK ] package tests

Distribution for gsl 1.22.0
Commit 55284ea320f1721e3e659d5814b7e3ce702e345b
Archive _build/gsl-1.22.0.tbz
dune-release: [ERROR] run ['dune-release' 'distrib' '--verbosity' 'warning']:
                      exited with 1
```

Any suggestions for a workaround until this is fixed?

--- a/lib/opam.ml
+++ b/lib/opam.ml
@@ -265,6 +265,24 @@ module Url = struct
         else (OS.File.must_exist distrib_file >>= fun _ -> assert false)
 end
 
+let opam_version () =
+  let v =
+    OS.Cmd.run_out Cmd.(cmd % "--version") |> OS.Cmd.out_string ~trim:true
+  in
+  let of_str = function
+  | "1.2.2" -> `v1_2_2
+  | s       ->
+      match String.cut ~sep:"2." s with
+      | Some ("", _) -> `v2
+      | _ -> Fmt.failwith "opam: invalid version %s" s
+  in
+  match v with
+  | Ok (v, (_, `Exited 0)) -> of_str v
+  | Ok (_, (_, s)) -> Fmt.failwith "opam: %a" OS.Cmd.pp_status s
+  | Error (`Msg e) -> Fmt.failwith "opam: %s" e
+
+let version = lazy (opam_version ())
+
 (*---------------------------------------------------------------------------
    Copyright (c) 2016 Daniel C. Bünzli
 

--- a/lib/opam.mli
+++ b/lib/opam.mli
@@ -97,6 +97,9 @@ module Url : sig
       the checksum of file [f]. *)
 end
 
+val version: [`v1_2_2 | `v2] Lazy.t
+(** [version] is the output of [opam --version]. *)
+
 (*---------------------------------------------------------------------------
    Copyright (c) 2016 Daniel C. Bünzli
 

--- a/lib/pkg.ml
+++ b/lib/pkg.ml
@@ -437,11 +437,10 @@ let lint_file_with_cmd ~dry_run file_kind ~cmd file errs handle_exit =
   |> Logs.on_error_msg ~use:(fun () -> errs + 1)
 
 let lint_opams ~dry_run p =
-  Logs.on_error_msg ~use:(fun () -> 1) (
-    (* remove opam.1.2-related warnings *)
-    opam_field p "opam-version" >>= fun opam_version ->
-    let args = match opam_version with
-    | Some ["1.2"] -> Cmd.v "--warn=-21-32-48"
+  let tool_version = Lazy.force Opam.version in
+  let lint opam_version =
+    let args = match opam_version, Lazy.force Opam.version with
+    | Some ["1.2"], `v2 -> Cmd.v "--warn=-21-32-48"
     | _ -> Cmd.empty
     in
     opam p >>= fun opam ->
@@ -454,10 +453,10 @@ let lint_opams ~dry_run p =
       ("" | "5" (* dirname version vs opam file version *)) -> `Ok
     | _ ->
         let err = OS.Cmd.err_run_out in
-        match
-          Sos.run_out ~dry_run ~err Cmd.(cmd % p file) ~default:(Sos.out "") OS.Cmd.out_string
-        with
-        | Ok (out, _) -> `Fail out
+        let cmd = Cmd.(cmd % p file)  in
+        let default = Sos.out "" in
+        match Sos.run_out ~dry_run ~err cmd ~default OS.Cmd.out_string with
+        | Ok (out, _     ) -> `Fail out
         | Error (`Msg out) -> `Fail out
     in
     let cmd = Cmd.(cmd % "-s") in
@@ -470,7 +469,18 @@ let lint_opams ~dry_run p =
       doc_user_repo_and_path p >>= fun _ ->
       distrib_user_and_repo p >>| fun _ ->
       d
-    ))
+    )
+  in
+  Logs.on_error_msg ~use:(fun () -> 1) (
+    (* remove opam.1.2-related warnings *)
+    opam_field p "opam-version" >>= fun opam_version ->
+    match opam_version, tool_version with
+    | Some ["2.0"], `v1_2_2 ->
+        Logs.app (fun m ->
+            m "Skipping opam lint as `opam-version` field is \"2.0\" \
+               while `opam --version` is 1.2.2");
+        Ok 0
+    | _ -> lint opam_version)
 
 type lint = [ `Std_files | `Opam ]
 

